src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -143,24 +154,35 @@ public void finishLocalSetup() {
 
         childContext = checkContext;
     }
-
+    /**
+     * {@inheritDoc} Creates child module.
+     * @noinspection ChainOfInstanceofChecks
+     */
     @Override
     public void setupChild(Configuration childConf)
             throws CheckstyleException {
         final String name = childConf.getName();
         final Object module = moduleFactory.createModule(name);
-        if (!(module instanceof AbstractCheck)) {
+        if (module instanceof AbstractCheck) {
+            final AutomaticBean bean = (AutomaticBean) module;
```

Comment:
@timurt 
Why do we need to treat AbstractCheck as AutomaticBean directly?  Please, restore original actions:
```java
        final AbstractCheck check = (AbstractCheck) module;
        check.contextualize(childContext);
        check.configure(childConf);
        check.init();

        registerCheck(check);
```